src/Microsoft.ML.LightGBM/LightGbmRankingTrainer.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.FastTree;
+using Microsoft.ML.Runtime.FastTree.Internal;
+using Microsoft.ML.Runtime.LightGBM;
+using Microsoft.ML.Runtime.Model;
+
+[assembly: LoadableClass(LightGbmRankingTrainer.Summary, typeof(LightGbmRankingTrainer), typeof(LightGbmArguments),
+    new[] { typeof(SignatureRankerTrainer), typeof(SignatureTrainer), typeof(SignatureTreeEnsembleTrainer) },
+    "LightGBM Ranking", LightGbmRankingTrainer.LoadNameValue, LightGbmRankingTrainer.ShortName, DocName = "trainer/LightGBM.md")]
+
+[assembly: LoadableClass(typeof(LightGbmRankingPredictor), null, typeof(SignatureLoadModel),
+    "LightGBM Ranking Executor",
+    LightGbmRankingPredictor.LoaderSignature)]
+
+namespace Microsoft.ML.Runtime.LightGBM
+{
+
+    public sealed class LightGbmRankingPredictor : FastTreePredictionWrapper
```

Comment:
do we need this line?

Reply:
yes

---
In reply to: [197550850](https://github.com/dotnet/machinelearning/pull/392#discussion_r197550850) [](ancestors = 197550850)